packages/component/src/hooks/internal/useMemoAll.spec.jsx

```diff
@@ -0,0 +1,146 @@
+    return new Promise(resolve => {
+      act(() => {
+        setTest(() => () => resolve(fun(...args)));
+      });
+    });
+  };
+};
+
+test('useMemoize should cache result across runs', async () => {
+  const expensiveSum = jest.fn((x, y) => x + y);
+
+  const render = testHook(doMemoChecks => {
+    // Start a run, all calls to sum() will be cached.
+    useMemoize(expensiveSum, sum => doMemoChecks(sum), []);
+  });
+
+  await render(sum => {
+    expect(sum(1, 2)).toBe(3); // Not cached, return 3.
+    expect(sum(1, 2)).toBe(3); // Cached, return 3.
+    expect(sum(2, 4)).toBe(6); // Not cached, return 6.
+    expect(sum(1, 2)).toBe(3); // Cached, return 3. This is cached because it is inside the same run.
+  });
+  expect(expensiveSum).toHaveBeenCalledTimes(2);
+
+  expect(expensiveSum.mock.calls[0]).toEqual([1, 2]);
+  expect(expensiveSum.mock.calls[1]).toEqual([2, 4]);
+
+  // After the run, 1 + 2 = 3, and 2 + 4 = 6 is cached.
+
+  // Start another run with previous cache
+  await render(sum => {
+    expect(sum(1, 2)).toBe(3); // Cached from previous run, return 3.
+    expect(sum(3, 6)).toBe(9); // Not cached, return 9.
+  });
+
+  expect(expensiveSum).toHaveBeenCalledTimes(3);
+  expect(expensiveSum.mock.calls[2]).toEqual([3, 6]);
+
+  // After the run, only 1 + 2 = 3 and 3 + 6 = 9 is cached. 2 + 4 is dropped.
+
+  // Start another run with previous cache
+  await render(sum => {
+    expect(sum(2, 4)).toBe(6); // Not cached, return 6
+  });
+
+  expect(expensiveSum).toHaveBeenCalledTimes(4);
+  expect(expensiveSum.mock.calls[3]).toEqual([2, 4]);
+});
+
+test('useMemoize should cache result if deps change', async () => {
+  const expensiveSum = jest.fn((x, y) => x + y);
+
+  const render = testHook(doMemoChecks => {
+    // Start a run, all calls to sum() will be cached.
+    useMemoize(expensiveSum, sum => doMemoChecks(sum), []);
+  });
+
+  // Start a run, all calls to sum() will be cached.
+  await render(sum => {
+    expect(sum(1, 2)).toBe(3); // Not cached, return 3.
+    expect(sum(1, 2)).toBe(3); // Cached, return 3.
+    expect(sum(2, 4)).toBe(6); // Not cached, return 6.
+    expect(sum(1, 2)).toBe(3); // Cached, return 3. This is cached because it is inside the same run.
+  });
+
+  expect(expensiveSum).toHaveBeenCalledTimes(2);
+
+  expect(expensiveSum.mock.calls[0]).toEqual([1, 2]);
+  expect(expensiveSum.mock.calls[1]).toEqual([2, 4]);
+
+  // After the run, 1 + 2 = 3, and 2 + 4 = 6 is cached.
+
+  // Start another run with previous cache
+  await render(sum => {
+    expect(sum(1, 2)).toBe(3); // Cached from previous run, return 3.
+    expect(sum(3, 6)).toBe(9); // Not cached, return 9.
+  });
+
+  expect(expensiveSum).toHaveBeenCalledTimes(3);
+  expect(expensiveSum.mock.calls[2]).toEqual([3, 6]);
+
+  // After the run, only 1 + 2 = 3 and 3 + 6 = 9 is cached. 2 + 4 is dropped.
+
+  // Start another run with previous cache
+  await render(sum => {
+    expect(sum(2, 4)).toBe(6); // Not cached, return 6
```

Comment:
Because 2 + 4 was dropped during previous run (1 + 2 and 3 + 6), it is inconclusive that 2 + 4 is not cached due to deps change.

Maybe try 1 + 2 or 3 + 6, which was not dropped.